### DISCUSSION TITLE
several errors during installation

### DISCUSSION THREAD

**Discussion by gustavoponciano** [Security-Onion-Solutions/securityonion#11234] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

6

### RAM

32GB

### Storage for /

162GB

### Storage for /nsm

327GB

### Network Traffic Collection

tap

### Network Traffic Speeds

more than 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hi Security Onion team,

I am trying to install Security Onion version 2.4.10 via ISO image on a CentOS 7 machine. I am getting the following errors:

```
Error: Unable to find a match: containerd.io-1.6.21-3.1.el9 docker-ce-24.0.4-1.el9 docker-ce-cli-24.0.4-1.el9 docker-ce-rootless-extras-24.0.4-1.el9
[ERROR   ] retcode: 1
[INFO    ] Executing command rpm in directory '/root'
[ERROR   ] Error occurred installing package(s). Additional info follows:

errors:
    - Running scope as unit run-18614.scope.
      Last metadata expiration check: 0:00:02 ago on Thu Aug 31 14:00:17 2023.
      No match for argument: containerd.io-1.6.21-3.1.el9
      No match for argument: docker-ce-24.0.4-1.el9
      No match for argument: docker-ce-cli-24.0.4-1.el9
      No match for argument: docker-ce-rootless-extras-24.0.4-1.el9
      Error: Unable to find a match: containerd.io-1.6.21-3.1.el9 docker-ce-24.0.4-1.el9 docker-ce-cli-24.0.4-1.el9 docker-ce-rootless-extras-24.0.4-1.el9


[WARNING ] 'version' argument will be ignored for multiple package targets
[INFO    ] Executing command systemd-run in directory '/root'
[ERROR   ] Command 'systemd-run' failed with return code: 1
[ERROR   ] stdout: Running scope as unit run-26592.scope.
Last metadata expiration check: 0:00:01 ago on Thu Aug 31 14:06:34 2023.
No match for argument: python3-dnf-plugin-versionlock
No match for argument: python3-docker
Package python36-m2crypto-0.35.2-5.el7.x86_64 is already installed.
Package python36-packaging-16.8-6.el7.noarch is already installed.
No match for argument: python3-pyyaml
No match for argument: python3-rich
No match for argument: python3-watchdog
Error: Unable to find a match: python3-dnf-plugin-versionlock python3-docker python3-pyyaml python3-rich python3-watchdog
[ERROR   ] retcode: 1
[INFO    ] Executing command rpm in directory '/root'
[ERROR   ] Error occurred installing package(s). Additional info follows:

errors:
    - Running scope as unit run-26592.scope.
      Last metadata expiration check: 0:00:01 ago on Thu Aug 31 14:06:34 2023.
      No match for argument: python3-dnf-plugin-versionlock
      No match for argument: python3-docker
      Package python36-m2crypto-0.35.2-5.el7.x86_64 is already installed.
      Package python36-packaging-16.8-6.el7.noarch is already installed.
      No match for argument: python3-pyyaml
      No match for argument: python3-rich
      No match for argument: python3-watchdog
      Error: Unable to find a match: python3-dnf-plugin-versionlock python3-docker python3-pyyaml python3-rich python3-watchdog
[INFO    ] Completed state [commonpkgs] at time 14:06:37.397886 (duration_in_ms=17055.074)
```

I have tried the following:

* I have updated the system repositories.
* I have tried installing the packages manually.
* I have tried using the `dnf config-manager` command to add the SaltStack repository.

I am still getting the same errors.

Can you please help me troubleshoot this issue?

Thanks,
Gustavo

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by biaeleasha** [comment]
I'm having the same issue.

**Reply by dougburks** [reply]
@biaeleasha Instead of replying to somebody else's discussion, please start your own discussion and provide details about your installation.

**Comment by dougburks** [comment]
@gustavoponciano Did you verify the ISO image? Do you get the same issue if you perform a fresh installation again?

**Reply by gustavoponciano** [reply]
The download and signature of the ISO image performed as expected. I've ran the installation a couple of times and it crashes and fails.

**Reply by defensivedepth** [reply]
`I am trying to install Security Onion version 2.4.10 via ISO image on a CentOS 7 machine. I am getting the following errors:`

Can you clarify this statement? Using the ISO and CentOS 7? CentOS 7 is not supported for 2.4.

**Reply by gustavoponciano** [reply]
I followed these steps:

https://github.com/Security-Onion-Solutions/securityonion/blob/2.4/main/DOWNLOAD_AND_VERIFY_ISO.md

My OS is CentOS 7.

**Reply by dougburks** [reply, marked as answer]
> My OS is CentOS 7.

Are you saying that you're using CentOS 7 to download and verify the ISO image? 

Or are you saying that you're trying to install Security Onion 2.4.10 on top of CentOS 7?

> I followed these steps:
https://github.com/Security-Onion-Solutions/securityonion/blob/2.4/main/DOWNLOAD_AND_VERIFY_ISO.md

After downloading and verifying the ISO image, are you then following the installation instructions at https://docs.securityonion.net/en/2.4/installation.html#installation-using-security-onion-iso-image to boot the machine from the Security Onion 2.4.10 ISO image and overwrite the CentOS 7 installation with the new Security Onion 2.4.10 installation?
